src/Microsoft.DotNet.Interactive.HttpRequestParser/Parser/HttpRequestParser.cs

```diff
@@ -322,30 +325,37 @@ private void ConsumeCurrentTokenInto(HttpSyntaxNode node)
             {
                 requestNode.Add(versionNode);
             }
-            else 
+            else
             {
-                ParseTrailingTrivia(requestNode, stopAfterNewLine: true);
+                ParseTrailingWhitespace(requestNode, stopAfterNewLine: true);
+            }
+
+            var commentsToPrepend = new List<HttpCommentNode>();
+            if (ParseComment() is { } comment)
+            {
+                commentsToPrepend.Add(comment);
             }
 
             var headersNode = ParseHeaders();
             if (headersNode is not null)
             {
+                PrependCommentsIfAny(commentsToPrepend, headersNode);
                 requestNode.Add(headersNode);
             }
 
-            var bodySeparatorNode = ParseBodySeparator();
-            if (bodySeparatorNode is not null)
-            {
-                requestNode.Add(bodySeparatorNode);
-            }
+            ParseTrailingWhitespace(requestNode);
 
             var bodyNode = ParseBody();
```

Comment:
> I think it may be better to let such comments be parsed by calling ParseLeadingWhiteSpaceAndComments() inside ParseBody().

I think this is confusing because the body can't have comments without an escaping mechanism, so I don't think we should be attaching comment nodes to body nodes, and I don't think we should be producing an empty body node based on a comment if we're a) not supporting body comments and b) not supporting empty bodies.